File: readability.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

int main(void)
{
    string input = get_string("What is your input? \n");

    int sentencecount = 0;
    for(int i = 0, j = strlen(input); i < j; i++ )
    {
        if(input[i] == '.' || input[i] == '!' || input[i] == '?')
        {
            sentencecount++;
        }
    }

    int wordcount = 0;
    bool ifword = false;
    for (int i = 0; input[i] != '\0'; i++ )
    {
         if(isspace(input[i]))
        {
            ifword = false;
        }
        else if(ifword == false)
        {
            wordcount++;
            ifword = true;
        }
    }

    int charcount = 0;
    bool ifchar = false;
    for (int i = 0; input[i] != '\0'; i++ )
    {
         if(isalpha(input[i]))
        {
            ifchar = true;
            charcount++;
        }
    }

    float l = (float)charcount / wordcount * 100;
    float s = (float)sentencecount / wordcount * 100;
    float index = .0588 * l - 0.296 * s - 15.8;
    int rounded_index = round(index);
    printf("Grade %i\n", rounded_index);
}

File: sentimental-readability/readability.py
string = input("What is your input? ")

charcount = 0
for i in range(len(string)):
    if string[i].isalpha():
        charcount = charcount + 1

wordcount = len(str.split(string))

sentencecount = string.count('.') + string.count('!') + string.count('?')

l = charcount / wordcount * 100
s = sentencecount / wordcount * 100
index = round(.0588 * l - 0.296 * s - 15.8)
if index < 1:
    print("Before Grade 1")
if index > 16:
    print("Grade 16+")
else:
    print(f"Grade {index}")
